### ISSUE TITLE
Test Public vs. Protected Endpoints With Ember Data

### ISSUE THREAD

**Issue by nadnoslen** [cybertooth-io/ermahgerd-rails-api-jwt#9]
_Originally posted by @nadnoslen in https://github.com/cybertooth-io/cppdist-com-rails/pull/3_

**Comment by nadnoslen** [comment]
I have tried playing around with nesting models in folders to mimic the `protected` route I've created but that's gone to hell in a hand-basket.  Namespace is not the answer.

Can try having everything under `api/v1/`.  So the `app/controllers/api/v1/roles_controller` would have a `app/resources/api/v1/role_resource` and the controller could inherit from `app/controllers/api/v1/protected_controller`.  The `ProtectedController` would have authenticate and authorize hooks embedded within.

Massive refactor on the way.
